tests/FakeItEasy.Tests/ServiceLocatorTests.cs

```diff
@@ -13,17 +13,25 @@ namespace FakeItEasy.Tests
     public class ServiceLocatorTests
     {
         [SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields", Justification = "Used reflectively.")]
-        private readonly Type[] singletonTypes =
+        private static IEnumerable<Type> SingletonTypes
         {
-            typeof(IExpressionCallMatcherFactory),
-            typeof(ExpressionArgumentConstraintFactory),
-            typeof(IProxyGenerator)
-        };
+            get
+            {
+                yield return typeof(IExpressionCallMatcherFactory);
+                yield return typeof(ExpressionArgumentConstraintFactory);
+                yield return typeof(IProxyGenerator);
+            }
+        }

```

Comment:
> It's not hugely important since this is just our test lib, but was there a problem with the original simple array?

Not really; but `SingletonTypes` was a field initialized with an array, and `NonSingletonTypes` was a property. I thought I'd make it consistent by making both of them properties.


> IEnumerable<T>+yield return is a beautiful language construct but it has it's costs, i.e. a lot of scaffolding and extra allocation in the background, so it's only beneficial in certain cases, e.g. if you may potentially yield a lot of items and consumers often bail out early for some reason, or if it results in much simpler code.

True, there was no real reason here. Since `NonSingletonTypes` was a property, it felt more natural to do it this way, then I did the same in `SingletonTypes` for consistency. I can change it to an array if you prefer.